crates/net/network/src/session/active.rs

```diff
@@ -757,6 +758,35 @@ fn calculate_new_timeout(current_timeout: Duration, estimated_rtt: Duration) ->
 
     smoothened_timeout.clamp(MINIMUM_TIMEOUT, MAXIMUM_TIMEOUT)
 }
+
+pub(crate) struct QueuedOutgoingMessages {
+    messages: VecDeque<OutgoingMessage>,
+    count: Gauge,
```

Comment:
looking at the ordering in Gaugefn impl for Atomic I believe this should work and not result in overflows